spec-tests/src/test/java/org/candlepin/spec/pools/PoolUnlimitedMasterSpecTest.java

```diff
@@ -0,0 +1,373 @@
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.StringUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+@SpecTest
+public class PoolUnlimitedMasterSpecTest {
+
+    private static ApiClient client;
+    private OwnerDTO owner;
+    private static OwnerClient ownerApi;
+    private static OwnerProductApi ownerProductApi;
+    private static ConsumerClient consumerApi;
+    private ConsumerDTO physicalSystem;
+    private ApiClient physicalClient;
+    private ConsumerDTO guest;
+    private ApiClient guestClient;
+    private ConsumerDTO guestUnmapped;
+    private ApiClient guestUnmappedClient;
+    private ProductDTO productNoVirt;
+    private ProductDTO productUnlimitedVirt;
+    private ProductDTO productVirt;
+    private ProductDTO proudctVirtHostDep;
+    private ProductDTO productVirtInstanceMuliplier;
+    private ProductDTO productVirtProductMuliplier;
+    private PoolDTO poolNoVirt;
+    private PoolDTO poolUnlimitedVirt;
+    private PoolDTO poolVirt;
+    private PoolDTO poolVirtHostDep;
+    private PoolDTO poolVirtInstanceMuliplier;
+    private PoolDTO poolVirtProductMuliplier;
+
+    @BeforeAll
+    public static void beforeAll() throws ApiException {
+        client = ApiClients.admin();
+        ownerApi = client.owners();
+        ownerProductApi = client.ownerProducts();
+        consumerApi = client.consumers();
+    }
+
+    @BeforeEach
+    public void beforeEach() throws ApiException, InterruptedException {
+        owner = ownerApi.createOwner(Owners.random());
+        String guestUuid = StringUtil.random("uuid");
+
+        physicalSystem = Consumers.random(owner)
+            .name(StringUtil.random("host"))
+            .type(ConsumerTypes.System.value())
+            .guestIds(List.of(new GuestIdDTO().guestId(guestUuid)));
+        physicalSystem = consumerApi.createConsumer(physicalSystem);
+        physicalClient = ApiClients.trustedConsumer(physicalSystem.getUuid());
+        guest = Consumers.random(owner)
+            .name(StringUtil.random("host"))
+            .type(ConsumerTypes.System.value())
+            .facts(Map.of("virt.uuid", guestUuid, "virt.is_guest", "true"));
+        guest = consumerApi.createConsumer(guest);
+        guestClient = ApiClients.trustedConsumer(guest.getUuid());
+        guestUnmapped = Consumers.random(owner)
+            .name(StringUtil.random("host"))
+            .type(ConsumerTypes.System.value())
+            .facts(Map.of("virt.is_guest", "true"));
+        guestUnmapped = consumerApi.createConsumer(guestUnmapped);
+        guestUnmappedClient = ApiClients.trustedConsumer(guestUnmapped.getUuid());
```

Comment:
We shouldn't be using trusted consumers (which are using the cp-consumer header in the background) for all of these clients. The ruby test is simply passing in the identity cert+key. The equivalent in java would be:
`ApiClients.ssl(guestUnmapped)`.

Which makes me wonder if we have other java spec tests doing the same (I'll check)